lib/workers/repository/update/branch/schedule.ts

```diff
@@ -99,43 +89,38 @@ export function hasValidSchedule(
   return [true];
 }
 
-function cronMatches(cron: string, now: DateTime, timezone?: string): boolean {
-  const parsedCron = parseCron(cron, timezone);
-
+export function cronMatches(
+  cron: string,
+  now: DateTime,
+  timezone?: string,
+): boolean {
+  const parsedCron: Cron = new Cron(cron, {
+    ...(timezone && { timezone }),
+  });
   // it will always parse because it is checked beforehand
   // istanbul ignore if
   if (!parsedCron) {
     return false;
   }
 
-  if (parsedCron.fields.hour.indexOf(now.hour as HourRange) === -1) {
-    // Hours mismatch
-    return false;
-  }
-
-  if (
-    parsedCron.fields.dayOfMonth.indexOf(now.day as DayOfTheMonthRange) === -1
-  ) {
-    // Days mismatch
+  // return the next date which matches the cron schedule
+  const nextRun = parsedCron.nextRun();
+  // istanbul ignore if: should not happen
+  if (!nextRun) {
+    logger.warn(`Invalid cron schedule ${cron}. No next run is possible`);
     return false;
   }
 
-  if (
-    !parsedCron.fields.dayOfWeek.includes(
-      (now.weekday % 7) as DayOfTheWeekRange,
-    )
-  ) {
-    // Weekdays mismatch
-    return false;
+  let nextDate: DateTime = DateTime.fromJSDate(nextRun);
```

Comment:
```suggestion
  let nextDate = DateTime.fromJSDate(nextRun);
```
I don't think we need the extra type declaration